### PR TITLE
Fix parsing of 0_ integer literals

### PR DESCRIPTION
**Pull request by turbolent**
Closes dapperlabs/cadence-private-issues#14
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/parser2/lexer/state.go [hunk 1]
```diff
@@ -210,7 +210,7 @@ func numberState(l *lexer) stateFn {
 			}
 			l.emitValue(TokenHexadecimalIntegerLiteral)
 
-		case '0', '1', '2', '3', '4', '5', '6', '7', '8', '9':
+		case '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '_':
```
Comment (SupunS) [inline runtime/parser2/lexer/state.go:213]: would this allow numbers starting with `_`? e.g: `_0`?

Comment (turbolent) [reply runtime/parser2/lexer/state.go:213]: No, this case only may occur after an initial `0`, see https://github.com/onflow/cadence/pull/1348/files#diff-e258ba2fee01ca54837c0062a323e801197f430aeff7ce6185b0130f51b511beR189 above

Comment (turbolent) [reply runtime/parser2/lexer/state.go:213]: Added in 71d7af0d0707dd41c745b2ba8c91c410b231626d

Comment (SupunS) [reply runtime/parser2/lexer/state.go:213]: Thank you!

**Pull request merged**